src/main/java/io/vertx/httpproxy/impl/ReverseProxy.java

```diff
@@ -35,13 +39,27 @@ public class ReverseProxy implements HttpProxy {
   public ReverseProxy(ProxyOptions options, HttpClient client) {
     CacheOptions cacheOptions = options.getCacheOptions();
     if (cacheOptions != null) {
-      Cache<String, Resource> cache = cacheOptions.newCache();
+      Cache cache = newCache(cacheOptions, ((HttpClientInternal) client).vertx());
       addInterceptor(new CachingFilter(cache));
     }
     this.client = client;
     this.supportWebSocket = options.getSupportWebSocket();
   }
 
+  public Cache newCache(CacheOptions options, Vertx vertx) {
+    if (options.getShared()) {
+      CloseFuture closeFuture = new CloseFuture();
+      return ((VertxInternal) vertx).createSharedResource("__vertx.shared.proxyCache", options.getName(), closeFuture, (cf_) -> {
+        Cache cache = new CacheImpl(options);
+        cf_.add(completion -> {
+          cache.close().onComplete(completion);
+        });
+        return cache;
+      });
+    }
+    return new CacheImpl(options);
```

Comment:
@tsegismont @wzy1935 

IIUC a custom cache implementation would implement the `Cache` interface (makes sense).

How do we tell the proxy to use that cache? Is there work still to be done on this?

Reply:
Yes, it is not possible based on this PR. There was a plan to add a Redis cache implementation later, by that time other custom cache implementations would be allowed